### DISCUSSION TITLE
Absolute IMPORT Feature

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#17111] [Feature Requests]
Hi everyone,
according to my understanding, currently imports are "relative" to current content of a collection:
- any import records matching a primary key are updating the data
- any import records not matching a primary key are creating a new data record

Is there any way to do an "absolute" import where:
- any import records matching a primary key are updating the data (no change)
- any import records not matching a primary key are creating a new data record (no change)
- ANY EXISTING RECORDS NOT BEING IN THE IMPORT SET ARE DELETED"

Thanks for advise,
Roman

**Comment by rijkvanzanten** [comment]
Just to be absolutely sure, you expect it to "upsert" (merge) with the existing data first, and remove any records that weren't updated/created? Or do you expect the whole table to be emptied first, and "set" to the file that you uploaded?

**Reply by ernst86913** [reply]
@rijkvanzanten  See below - seems GitHub mail responses don’t end up directly in the originating thread… and the formatting is …. Bad

**Comment by ernst86913** [comment]
Hi Rijk -To be honest - not sure what the right approach would be… maybe best to describe a use case:Let’s say the management of a product price list is done in an XLS… and it’s updated once a week. With every update it is uploaded to directus.Key therefore is:1. After the update the records in Directus need to exactly match the records of the XLS2. Unique keys need to be maintained to avoid losing and addtl data which are in directus and to maintain any relationships between the product and the pricelist schemas3. Any records which are deleted in the XLS should be deleted4. Any new record in the XLS should be created5. Any updated records in the XLS should be updatedDifference to current import:- current import is a relative one… records are created and update- but - records which are not in the XLS anymore stay in directus. In above use case this means that prices would still be available for purchase… which is not goodThat’s why I’d propose to also have the option for an „absolute“ import.BTW - this could also be perfectly used for migrations of permissions, roles, flows,… from one environment to the other …

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
